### PR TITLE
Fix IBMQ.get_backends and IBMQ.backends for new backend names

### PR DESCRIPTION
**Pull request by nonhermitian**
### Summary
The default IBMQ backend names were changed to their city names, causing `IBMQ.get_backend` and `IBMQ.backend('name')` to fail with backend not found.  This fixes that by moving the qx names to depreciated, this making aliased names an empty dict.  This also includes a test for explicitly getting both city and qx backends if they are in the depreciated names list.

### CODE REVIEW

File: qiskit/backends/providerutils.py [hunk 1]
```diff
@@ -41,6 +41,14 @@ def _match_all(dict_, criteria):
     status_filters = {}
     for key, value in kwargs.items():
         if all(key in backend.configuration() for backend in backends):
+            if key == 'name':
```
Comment (diego-plan9) [inline qiskit/backends/providerutils.py]: Can you revert this change, and introduce back the 
```
# Special handling of the `name` parameter, to support alias resolution.
```
block in `ibmqprovider.py`? This `_match_all()` function should be general and not need knowledge of the ibmq internals, and will probably end up causing circular import problems plus breaking the intended modularization.

Comment (nonhermitian) [reply qiskit/backends/providerutils.py]: Indeed, that removal was done before I realized that moving to depreciated backnames is the thing to do.

Comment (diego-plan9) [reply qiskit/backends/providerutils.py]: Cool, thanks for the speedy update!

**Comment by nonhermitian** [comment]
The issue is now that even if the backend can be found, the device cannot be found when doing execute (works fine for backends such as melbourne that are not depreciated /aliased):

```python

from qiskit import *
from qiskit import IBMQ

IBMQ.load_accounts()

backend = IBMQ.get_backend('ibmq_5_tenerife')

q = QuantumRegister(2)
c = ClassicalRegister(2)
qc = QuantumCircuit(q, c)
qc.h(q[0])
qc.cx(q[0], q[1])
qc.measure(q, c)

job = execute(qc, backend=backend)
```

```
{"status":404,"message":"Device not found.","code":"DEVICE_NOT_FOUND","statusCode":404}}
```

**Comment by muneerqu** [comment]
facing the same/similar issue on 0.5.7

```python
for backend in backends:
    pprint(backend.status)
```
```python
[<qiskit.backends.ibmq.ibmqbackend.IBMQBackend object at 0x7f5cf42a3dd8>,
 <qiskit.backends.ibmq.ibmqbackend.IBMQBackend object at 0x7f5cf42a3710>,
 <qiskit.backends.ibmq.ibmqbackend.IBMQBackend object at 0x7f5cf42a3d68>,
 <qiskit.backends.ibmq.ibmqbackend.IBMQBackend object at 0x7f5cf42a3b38>,
 <qiskit.backends.ibmq.ibmqbackend.IBMQBackend object at 0x7f5d31474b38>,
 <qiskit.backends.local.qasm_simulator_cpp.QasmSimulatorCpp object at 0x7f5cf433ca20>,
 <qiskit.backends.local.statevector_simulator_cpp.StatevectorSimulatorCpp object at 0x7f5cf42f69e8>,
 <qiskit.backends.local.unitary_simulator_py.UnitarySimulatorPy object at 0x7f5cf42f6a58>]
```
```python
for backend in backends:
    pprint(backend.name)
```

```python
'ibmq_16_melbourne'
'ibmq_qasm_simulator'
'ibmq_5_yorktown'
'ibmq_5_tenerife'
'ibmq_16_rueschlikon'
'local_qasm_simulator_cpp'
'local_statevector_simulator_cpp'
'local_unitary_simulator_py'
```

```python
for backend in backends:
    pprint(backend.status)
```

```
{'pending_jobs': 179, 'name': 'ibmq_16_melbourne', 'operational': True}
{'pending_jobs': 0, 'name': 'ibmq_qasm_simulator', 'operational': True}

---------------------------------------------------------------------------
BadBackendError                           Traceback (most recent call last)
~/anaconda3/envs/envsQISkit/lib/python3.6/site-packages/qiskit/backends/ibmq/ibmqbackend.py in status(self)
    149             backend_name = self._name_for_api
--> 150             status = self._api.backend_status(backend_name)
    151             # FIXME a hack to rename the key. Needs to be fixed in api

~/anaconda3/envs/envsQISkit/lib/python3.6/site-packages/IBMQuantumExperience/IBMQuantumExperience.py in backend_status(self, backend, access_token, user_id)
    882         if not backend_type:
--> 883             raise BadBackendError(backend)
    884 

BadBackendError: Could not find backend "ibmqx2" available.

During handling of the above exception, another exception occurred:

LookupError                               Traceback (most recent call last)
<ipython-input-4-ba9e747b415d> in <module>()
      1 for backend in backends:
----> 2     pprint(backend.status)

~/anaconda3/envs/envsQISkit/lib/python3.6/site-packages/qiskit/backends/ibmq/ibmqbackend.py in status(self)
    169         except Exception as ex:
    170             raise LookupError(
--> 171                 "Couldn't get backend status: {0}".format(ex))
    172         return AvailableToOperationalDict(status)
    173 

LookupError: Couldn't get backend status: Could not find backend "ibmqx2" available.
```

QISKIT: 0.5.7
Python: 3.6.6 Anaconda
System: Ubuntu 18.04 LTS

**Comment by nonhermitian** [comment]
This is no longer needed.
